src/modules/schnorrsig/main_impl.h

```diff
@@ -314,12 +306,19 @@ int secp256k1_schnorrsig_verify_batch(const secp256k1_context *ctx, secp256k1_sc
     VERIFY_CHECK(ctx != NULL);
     ARG_CHECK(secp256k1_ecmult_context_is_built(&ctx->ecmult_ctx));
     ARG_CHECK(scratch != NULL);
+    /* Check that n_sigs is less than half of the maximum size_t value. This is necessary because
+     * the number of points given to ecmult_multi is 2*n_sigs. */
+    ARG_CHECK(n_sigs < (size_t)1 << (sizeof(size_t)*8-1));
```

Comment:
`1 << sizeof(size_t)*8-1` is not necessarily half of the maximum value that can be represented by size_t. For example, there could be unused "padding bits" in the integer. This is only a problem on exotic platforms that we don't support anyway but yeah... Anyway,  a simpler expression is `MAX_SIZE >> 1` (only in C99) or equivalently `(size_t)(-1) >> 1` because we want C89.

BUT I don't know if we need to go this far, I guess there are a lot of places where the code anyway assumes that integers are at least 32 bits. So we probably can just use the check below? @sipa

Reply:
I would prefer `(size_t)(-1) >> 1` because it's easier to read. Or even use `/ 2` instead of `>> 1`, this will be evaluated at compile-time anyway.

Reply:
fixed